[CI] ClientYamlTestSuiteIT class failing

**Build Scans:**
- [elasticsearch-periodic #3851 / openjdk22_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/6maoxteklillo)
- [elasticsearch-pull-request #29074 / part-1](https://gradle-enterprise.elastic.co/s/k6mhgduvn43y4)
- [elasticsearch-pull-request #29051 / rest-compatibility](https://gradle-enterprise.elastic.co/s/hotuur3x3sjaa)
- [elasticsearch-pull-request #29043 / part-1](https://gradle-enterprise.elastic.co/s/psswu2oubecj6)
- [elasticsearch-pull-request #29043 / rest-compatibility](https://gradle-enterprise.elastic.co/s/er56noamzqecc)
- [elasticsearch-pull-request #29009 / part-1](https://gradle-enterprise.elastic.co/s/4kxpz2tqijke4)
- [elasticsearch-pull-request #28969 / part-1](https://gradle-enterprise.elastic.co/s/jg4jh7dmsmfay)
- [elasticsearch-pull-request #28932 / part-1](https://gradle-enterprise.elastic.co/s/6t7ceg2lbc5jw)
- [elasticsearch-pull-request #28896 / part-1](https://gradle-enterprise.elastic.co/s/7ief4al7prpyy)
- [elasticsearch-pull-request #28773 / part-1](https://gradle-enterprise.elastic.co/s/xdaocttcqmyrq)

**Reproduction Line:**
```
./gradlew ":rest-api-spec:yamlRestTest" --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT.test {yaml=index/92_metrics_auto_subobjects/Metrics object indexing with synthetic source}" -Dtests.seed=E99EE906E57CF0A -Dtests.locale=sk -Dtests.timezone=Asia/Ust-Nera -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.ClientYamlTestSuiteIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.ClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [index/92_metrics_auto_subobjects:133]: got unexpected warning header [
	299 Elasticsearch-8.16.0-2b1170509b39b5a23c00bbfa9da87144868a5df2 "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
]

```

**Issue Reasons:**
- [main] 11 failures in class org.elasticsearch.test.rest.ClientYamlTestSuiteIT (1.2% fail rate in 907 executions)
- [main] 2 failures in step rest-compatibility (1.3% fail rate in 152 executions)
- [main] 8 failures in step part-1 (5.5% fail rate in 145 executions)
- [main] 9 failures in pipeline elasticsearch-pull-request (5.6% fail rate in 161 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/91_metrics_no_subobjects.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/91_metrics_no_subobjects.yml
@@ -6,6 +6,8 @@
       reason: added in 8.3.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -71,6 +73,8 @@
       reason: added in 8.3.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -131,6 +135,8 @@
       reason: added in 8.4.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -199,6 +205,8 @@
       reason: added in 8.4.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/91_metrics_no_subobjects.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/91_metrics_no_subobjects.yml
@@ -1,7 +1,7 @@
 ---
 "Metrics object indexing":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["gte_v8.3.0"]
       reason: added in 8.3.0
 
@@ -68,7 +68,7 @@
 ---
 "Root without subobjects":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["gte_v8.3.0"]
       reason: added in 8.3.0
 
@@ -130,7 +130,7 @@
 ---
 "Metrics object indexing with synthetic source":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["gte_v8.4.0"]
       reason: added in 8.4.0
 
@@ -200,7 +200,7 @@
 ---
 "Root without subobjects with synthetic source":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["gte_v8.4.0"]
       reason: added in 8.4.0
 

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/92_metrics_auto_subobjects.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/92_metrics_auto_subobjects.yml
@@ -6,6 +6,8 @@
       reason: requires supporting subobjects auto setting
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -71,6 +73,8 @@
       reason: requires supporting subobjects auto setting
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -131,6 +135,8 @@
       reason: added in 8.4.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -199,6 +205,8 @@
       reason: added in 8.4.0
 
   - do:
+      allowed_warnings:
+        - "index template [test] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/92_metrics_auto_subobjects.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/index/92_metrics_auto_subobjects.yml
@@ -1,7 +1,7 @@
 ---
 "Metrics object indexing":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["mapper.subobjects_auto"]
       reason: requires supporting subobjects auto setting
 
@@ -68,7 +68,7 @@
 ---
 "Root with metrics":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["mapper.subobjects_auto"]
       reason: requires supporting subobjects auto setting
 
@@ -130,7 +130,7 @@
 ---
 "Metrics object indexing with synthetic source":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["mapper.subobjects_auto"]
       reason: added in 8.4.0
 
@@ -200,7 +200,7 @@
 ---
 "Root without subobjects with synthetic source":
   - requires:
-      test_runner_features: allowed_warnings_regex
+      test_runner_features: [ "allowed_warnings", "allowed_warnings_regex" ]
       cluster_features: ["mapper.subobjects_auto"]
       reason: added in 8.4.0
 

